modules/beta-autopilot-private-cluster/variables.tf

```diff
@@ -0,0 +1,522 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// This file was automatically generated from a template in ./autogen/main
+
+variable "project_id" {
+  type        = string
+  description = "The project ID to host the cluster in (required)"
+}
+
+variable "name" {
+  type        = string
+  description = "The name of the cluster (required)"
+}
+
+variable "description" {
+  type        = string
+  description = "The description of the cluster"
+  default     = ""
+}
+
+variable "regional" {
+  type        = bool
+  description = "Whether is a regional cluster (zonal cluster if set false. WARNING: changing this after cluster creation is destructive!)"
+  default     = true
+}
+
+variable "region" {
+  type        = string
+  description = "The region to host the cluster in (optional if zonal cluster / required if regional)"
+  default     = null
+}
+
+variable "zones" {
+  type        = list(string)
+  description = "The zones to host the cluster in (optional if regional cluster / required if zonal)"
+  default     = []
+}
```

Comment:
Since ap is only regional, we can just have region?

Reply:
working this out would either cause a split in logic {if statement based on autopilot_cluster} or a rewrite of the  [locals in the main.tf](https://github.com/cmcga1125/terraform-google-kubernetes-engine/blob/cmcga1125/835/modules/beta-autopilot-private-cluster/main.tf#L36) file.

another more simple solution would be to set the variable regional to `true` then remove it from the examples/documentation for this cluster?  What do ya'll think?